sunrise.py

```diff
@@ -0,0 +1,379 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+# coordinates for solar farm 1.0
+lat = 40.081798
+lon = -88.244027
+utc = -6
+
+
+def timestamp_to_hour(timestamp):
+    """
+    This function returns the hour number
+    of a particular timestamp.
+
+    Parameters:
+    -----------
+    timestamp : pandas Timestamp
+        The time stamp of interest, must be in 24-hour.
+
+    Returns:
+    --------
+
+    """
+    minutes_of_hour = timestamp.minute / 60
+    seconds_of_hour = timestamp.second / 3600
+    hour_frac = timestamp.hour + minutes_of_hour + seconds_of_hour
+    day_number = timestamp.dayofyear
+
+    hour = hour_number(day_number, hour_frac)
+
+    return hour
+
+
+def hour_number(N, time):
+    """
+    Takes the day number and time (in hours) and
+    converts to hour number.
+
+    Parameters:
+    -----------
+    N : integer
+        The day number
+    time : float
+        The time in hours (24-hour clock)
+
+    Returns:
+    --------
+    hour : float
+        The hour number
+    """
+    hour = N * 24 + time
+
+    return hour
+
+
+def day_number(hour_number):
+    """
+    Returns the day_number given a particular hour_number.
+    """
+    N = (hour_number / 8760) * 365
+
+    return int(N)
+
+
+def local_time(hour_number):
+    """
+    Takes the hour number, and converts to local time.
+    """
+    N = day_number(hour_number)
+
+    time = hour_number - N * 24
+
+    return time
+
+
+def frac_year(hour, leap_year=False):
+    """
+    This function calculates the fraction of the year.
+
+    Parameters:
+    -----------
+    hour : integer, float
+        The hour of the day. Could be a decimal value.
+    leap_year : boolean
+        Indicates if the year is a leap year. Default
+        is False.
+
+    Returns:
+    --------
+    B : float
+        The fraction of the year
+    """
+    if leap_year:
+        n_days = 366
+    else:
+        n_days = 365
+
+    B = (hour - 1944) / 24 * 360 / n_days
+    return B
+
+
+def declination(hour, leap_year=False):
+    """
+    This function calculates the declination angle of the sun.
+    Parameters:
+    -----------
+    hour : integer, float
+        The hour of the day. Could be a decimal value.
+    leap_year : boolean
+        Indicates if the year is a leap year. Default
+        is False.
+
+    Returns:
+    --------
+    delta : float
+        The declination at the given hour
+    """
+    B = frac_year(hour, leap_year)
+    delta = 23.44 * np.sin((np.pi / 180) * B)
+
+    return delta
+
+
+def equation_of_time(hour, leap_year=False):
+    """
+    This function calculates the equation of time. The equation
+    of time gives the difference between the solar time and wall
+    clock time. Effectively correcting for the eccentricity of
```

Comment:
I think it should be "time, effectively" rather than "time. Effectively"

Reply:
```suggestion
    clock time, effectively correcting for the eccentricity of
```